app/assets/javascripts/discourse/lib/url.js.es6

```diff
@@ -187,13 +187,7 @@ const DiscourseURL = Ember.Object.extend({
     const pathname = path.replace(/(https?\:)?\/\/[^\/]+/, "");
     const baseUri = Discourse.BaseUri;
 
-    // If we have a baseUri and an absolute URL, make sure the baseUri
-    // is the same. Otherwise we could be switching forums.
-    if (
-      baseUri &&
-      path.indexOf("http") === 0 &&
-      pathname.indexOf(baseUri) !== 0
-    ) {
+    if (!DiscourseURL.isInternal(path)) {
```

Comment:
Are there tests ensuring this is the same behavior?

Reply:
Hmm... I think this might behave different in subdirectory installations.

Reply:
Yeah, you definitely don't want to break subfolder support ;)

Reply:
I tested this on a subfolder setup and looks good to me.